sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/generators.scala

```diff
@@ -431,6 +432,22 @@ case class Explode(child: Expression) extends ExplodeBase {
     copy(child = newChild)
 }
 
+trait ExplodeGeneratorBuilderBase extends GeneratorBuilder {
+  override def functionSignatures: Option[Seq[FunctionSignature]] =
+    Some(Seq(FunctionSignature(Seq(NamedArgument("collection")))))
```

Comment:
Does it have to be defined like this? Can this be simplified?

Reply:
To best explain this:

1. The option is used to check if named arguments will be supported for this expression builder.
2. The seq is potentially useful in the future for function overloads.
3. There can be additional information that can be packaged into the FunctionSignature in the future.
4. NamedArguments can hold additional information like default values. 

So most of these things are pretty useful. For now, I think it makes sense to keep as is.